### ISSUE TITLE
PHP 8.1 Problem with Response (Array/JSON)

### ISSUE THREAD

**Issue by mdobydullah** [laravel/framework#42681]
I've updated the PHP version to 8.1.6. Then it doesn't return full response. Here's an example:

Code:
```php
$data = [
    'success' => true,
    'message' => 'Updated'
];
return response()->json($data);
```

Response:
```

{"success":true,"message":"Updated
```

In the response, it includes a 'new line' first, then it shows incomplete response (missing curly bracket at the end).

I'm using: PHP 8.1.6 and Laravel 9.16.0.

Please help me to resolve the issue. Thank you.

**Comment by eusonlito** [comment]
Same issue? https://github.com/laravel/framework/issues/42654

**Comment by mdobydullah** [comment]
@eusonlito 

Yes. Same issue.

**Comment by rodrigopedra** [comment]
@mdobydullah can you provide a public repository with code that we can reproduce the issue?

For reference on how to do it you can refer to this comment: https://github.com/laravel/framework/issues/42654#issuecomment-1146757460

**Comment by mdobydullah** [comment]
@eusonlito @rodrigopedra

I've downgraded "**symfony/http-foundation**" version to "**6.0.9**" and it has resolved the issue.

```
"require": {
    ---
    "symfony/http-foundation": "6.0.*"
},
```

It is a temp solution.

**Comment by eusonlito** [comment]
`symfony/http-foundation` has commit currently reverted adding the `Content-Length` header. Maybe all problems came frome here: https://github.com/symfony/http-foundation/commit/a58dc88d56e04e57993d96c1407a17407610e1df

**Comment by mdobydullah** [comment]
@eusonlito Yh. I think so.

**Comment by rodrigopedra** [comment]
And that might be the reason it is hard to replicate, from the comments on this issue: https://github.com/symfony/symfony/issues/46449 it only happens under particular conditions (Apache server, gzip enabled).

**Comment by driesvints** [comment]
Seems this is a Symfony issue. Please don't upgrade to Symfony v6.1 for now and wait until they release a patch release.

**Comment by mdobydullah** [comment]
> `symfony/http-foundation` release `6.1.0` has commit currently reverted adding the `Content-Length` header. Maybe all problems came frome here: [symfony/http-foundation@a58dc88](https://github.com/symfony/http-foundation/commit/a58dc88d56e04e57993d96c1407a17407610e1df)

Hi @eusonlito @rodrigopedra @driesvints,

FYI, Symfony has fixed the issue and released ```symfony/http-foundation: v6.1.1```.

https://github.com/symfony/http-foundation/releases/tag/v6.1.1

Thank you so much for your help.
